ci/build_wheel.sh

```diff
@@ -3,20 +3,50 @@
 
 set -euo pipefail
 
+package_name="cuml"
+package_dir="python"
+
 source rapids-configure-sccache
 source rapids-date-string
 
 # Use gha-tools rapids-pip-wheel-version to generate wheel version then
 # update the necessary files
-version_override="$(rapids-pip-wheel-version ${RAPIDS_DATE_STRING})"
+version_override=$(rapids-generate-version)
+commit_override=$(git rev-parse HEAD)
 
 RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
 
-ci/release/apply_wheel_modifications.sh ${version_override} "-${RAPIDS_PY_CUDA_SUFFIX}"
-echo "The package name and/or version was modified in the package source. The git diff is:"
-git diff
-
-cd python/
+# This is the version of the suffix with a preceding hyphen. It's used
+# everywhere except in the final wheel name.
+PACKAGE_CUDA_SUFFIX="-${RAPIDS_PY_CUDA_SUFFIX}"
+
+# Patch project metadata files to include the CUDA version suffix and version override.
+pyproject_file="${package_dir}/pyproject.toml"
+init_file="${package_dir}/${package_name}/__init__.py"
+
+sed -i "s/^name = \"cuml\"/name = \"cuml${PACKAGE_CUDA_SUFFIX}\"/g" ${pyproject_file}
+sed -i "s/__version__ = .*/__version__ = ${version_override}/g" ${init_file}
+sed -i "s/__git_commit__ = .*/__git_commit__ = \"${commit_override}\"/g" ${init_file}
+
+# For nightlies we want to ensure that we're pulling in alphas as well. The
+# easiest way to do so is to augment the spec with a constraint containing a
+# min alpha version that doesn't affect the version bounds but does allow usage
+# of alpha versions for that dependency without --pre
+alpha_spec=''
+if ! rapids-is-release-build; then
+    alpha_spec=',>=0.0.0a0'
+fi
+
+for dep in cudf pylibraft raft-dask rmm; do
+    sed -r -i "s/${dep}==(.*)\"/${dep}${PACKAGE_CUDA_SUFFIX}==\1${alpha_spec}\"/g" ${pyproject_file}
+done
+
+if [[ $PACKAGE_CUDA_SUFFIX == "-cu12" ]]; then
+    sed -i "s/cuda-python[<=>\.,0-9]*/cuda-python>=12.0,<13.0/g" ${pyproject_file}
+    sed -i "s/cupy-cuda11x/cupy-cuda12x/g" ${pyproject_file}
+fi
```

Comment:
This looks brittle.

Reply:
It is. Unfortunately, it's the best we can do at the moment. The long-term solution we want is https://github.com/rapidsai/dependency-file-generator/pull/48, however that work has been stalled for a while. Ultimately we definitely want to be able to generate CUDA 12 dependencies for wheels properly rather than perform brittle textual replacement.

Note that these commands aren't new, it's just been moved from `apply_wheel_modifications.sh` (deleted in this PR) to simplify the pipeline and reduce the amount of indirection. Equivalent changes were made to the rest of RAPIDS a while ago but cuml got lost in the mix because CI was blocked on this repo for a while.